app/renderer/abr-document.js

```diff
@@ -319,23 +332,26 @@ AbrDocument.prototype = {
         if (!path) {
             return this.saveAs(callback);
         }
+        // Pause the watcher to avoid triggering callbacks after saving the document
+        this.pauseWatcher();
         var that = this,
-            data = this.getData(),
-            callback2 = function (err) {
-                if (err) {
-                    return dialogs.fileAccessDenied(path, function () {
-                        that.saveAs(callback);
-                    });
-                }
-                that.setClean();
-                that.setPath(path);
-                that.ipcClient.trigger("setWinPath", path);
-                that.updateWindowTitle();
-                if (typeof callback === "function") {
-                    callback();
-                }
-            };
-        return files.writeFile(data, path, callback2);
+            data = this.getData();
+        files.writeFile(data, path, function (err) {
+            if (err) {
```

Comment:
You're right; the watcher won't restart if the user cancels here.

But it looks weird if I just call `that.startWatcher()` twice in the same function.
It would be clearer if `dialogs.fileAccessDenied()` can return the user's choice instead of always `false`.
Then the additional call to `that.startWatcher()` would be justified in the code.

What do you think?